File: AimPlugin3.0.4/AIM.Annotation/Template/CharacteristicQuantification.cs
//L  
// Copyright Northwestern University
// Copyright Stanford University (ATB 1.0 and ATS 1.0)
//
// Distributed under the OSI-approved BSD 3-Clause License.
// See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.

using System.Xml.Serialization;
using System.Collections.Generic;

namespace AIM.Annotation.Template
{
	public enum CharacteristicQuantificationType
	{
		Numerical,
		Quantile,
		NonQuantifiable,
		Scale,
		Interval
	}

	public class CharacteristicQuantification
	{
		public CharacteristicQuantification()
		{
			NonQuantifiable = new List<NonQuantifiable>();
			Interval = new List<Interval>();
			Numerical = new List<Numerical>();
		}

		[XmlAttribute("name")]
		public string Name { get; set; }

		[XmlAttribute("annotatorConfidence")]
		public bool AnnotatorConfidence { get; set; }

		[XmlElement("Scale", Namespace = TemplateContainer.Namespace)]
		public Scale Scale { get; set; }

		[XmlElement("Numerical", Namespace = TemplateContainer.Namespace)]
		public List<Numerical> Numerical { get; set; }

		[XmlElement("Interval", Namespace = TemplateContainer.Namespace)]
		public List<Interval> Interval { get; set; }

		[XmlElement("Quantile", Namespace = TemplateContainer.Namespace)]
		public Quantile Quantile { get; set; }

		[XmlElement("NonQuantifiable", Namespace = TemplateContainer.Namespace)]
		public List<NonQuantifiable> NonQuantifiable { get; set; }
	}
}

File: AIMToolkit_v3.0.2_rv11/source/AIMLib/AIMLib/model/CharacteristicQuantification.cpp
/*
*  Copyright Northwestern University
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/


#include "../stdafx.h"

#include "AimBase.h"
#include "CharacteristicQuantification.h"

#include "../AimEnum.h"
#include "Numerical.h"
#include "Quantile.h"
#include "NonQuantifiable.h"
#include "Scale.h"
#include "Interval.h"

#include "../memdebug.h"

using std::string;
using namespace aim_lib;

CharacteristicQuantification::CharacteristicQuantification(const string& typeName) : AimBase(typeName)
{
}

CharacteristicQuantification::CharacteristicQuantification(const CharacteristicQuantification& characteristicQuantification) : AimBase(characteristicQuantification)
{
	*this = characteristicQuantification;
}

CharacteristicQuantification& CharacteristicQuantification::operator=(const CharacteristicQuantification& characteristicQuantification)
{
	if (this != &characteristicQuantification)
	{
		AimBase::operator=(characteristicQuantification);
		_cagridId	= characteristicQuantification._cagridId;
		_name		= characteristicQuantification._name;

		SetAnnotatorConfidence(characteristicQuantification.GetAnnotatorConfidence());
	}
	return *this;
}

CharacteristicQuantification::~CharacteristicQuantification(void)
{
}

const string& CharacteristicQuantification::GetName() const
{
	return _name;
}

double* CharacteristicQuantification::GetAnnotatorConfidence() const
{
	return _annotatorConfidence.get();
}

void CharacteristicQuantification::SetName(const string& newVal)
{
	_name = newVal;
}

void CharacteristicQuantification::SetAnnotatorConfidence(double* pConfidence)
{
	if (pConfidence == NULL)
		_annotatorConfidence.reset();
	else
		_annotatorConfidence.reset(new double(*pConfidence));
}

const CharacteristicQuantification::E_QuantificationType CharacteristicQuantification::GetQuantificationType() const
{
	if (dynamic_cast<const Numerical*>(this) != NULL)
		return QT_Numerical;
	else if(dynamic_cast<const Quantile*>(this) != NULL)
		return QT_Quantile;
	else if(dynamic_cast<const NonQuantifiable*>(this) != NULL)
		return QT_Non_Quantifiable;
	else if(dynamic_cast<const Scale*>(this) != NULL)
		return QT_Scale;
	else if(dynamic_cast<const Interval*>(this) != NULL)
		return QT_Interval;
	else
		return QT_Unknown;
}
